src/main/java/io/kestra/plugin/shopify/orders/CreateOrder.java

```diff
@@ -0,0 +1,350 @@
+            orderData.put("tags", tagsValue);
+        }
+        }
+
+        if (currency != null) {
+        String currencyValue = runContext.render(currency).as(String.class).orElse(null);
+        if (currencyValue != null) {
+            orderData.put("currency", currencyValue);
+        }
+        }
+
+        // Addresses
+        if (billingAddress != null) {
+        AddressInput billingAddressValue = runContext.render(billingAddress).as(AddressInput.class).orElse(null);
+        if (billingAddressValue != null) {
+            orderData.put("billing_address", convertAddressToMap(billingAddressValue));
+        }
+        }
+
+        if (shippingAddress != null) {
+        AddressInput shippingAddressValue = runContext.render(shippingAddress).as(AddressInput.class).orElse(null);
+        if (shippingAddressValue != null) {
+            orderData.put("shipping_address", convertAddressToMap(shippingAddressValue));
+        }
+        }
+
+        // Inventory behaviour
+        String inventoryBehaviourValue = runContext.render(inventoryBehaviour).as(String.class).orElse("bypass");
+        orderData.put("inventory_behaviour", inventoryBehaviourValue);
+
+        // Send receipt flags
+        Boolean sendReceiptValue = runContext.render(sendReceipt).as(Boolean.class).orElse(false);
+        Boolean sendFulfillmentReceiptValue = runContext.render(sendFulfillmentReceipt).as(Boolean.class).orElse(false);
+        orderData.put("send_receipt", sendReceiptValue);
+        orderData.put("send_fulfillment_receipt", sendFulfillmentReceiptValue);
+
+        Map<String, Object> requestBody = Map.of("order", orderData);
+
+        URI uri = buildApiUrl(runContext, "/orders.json");
+        HttpRequest request = buildAuthenticatedRequest(runContext, "POST", uri, requestBody);
+
+        runContext.logger().debug("Creating order in Shopify API: {}", uri);
+        
+        handleRateLimit();
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        Map<String, Object> responseData = parseResponse(response);
+        
+        @SuppressWarnings("unchecked")
+        Map<String, Object> createdOrderData = (Map<String, Object>) responseData.get("order");
+        
+        if (createdOrderData == null) {
+        throw new RuntimeException("Failed to create order - no order data returned");
+        }
+        
+        Order createdOrder = OBJECT_MAPPER.convertValue(createdOrderData, Order.class);
+
+        runContext.logger().info("Created order '{}' (ID: {}) in Shopify", 
+            createdOrder.getName(), createdOrder.getId());
+
+        return Output.builder()
+            .order(createdOrder)
+            .build();
+    }
+
+    private Map<String, Object> convertAddressToMap(AddressInput address) {
+        Map<String, Object> addressMap = new HashMap<>();
+        
+        if (address.getFirstName() != null) addressMap.put("first_name", address.getFirstName());
```

Comment:
Same comment than for the other class here you can simplify it for readability with an helper method